llama_index/embeddings/nomic.py

```diff
@@ -0,0 +1,85 @@
+from llama_index.core.embeddings.base import DEFAULT_EMBED_BATCH_SIZE, BaseEmbedding
+from typing import Optional, List, Any
+from llama_index.bridge.pydantic import Field, PrivateAttr
+from llama_index.callbacks import CallbackManager
+from enum import Enum
+
+class NomicAITaskType(str, Enum):
+    SEARCH_QUERY = "search_query"
+    SEARCH_DOCUMENT = "search_document"
+    CLUSTERING = "clustering"
+    CLASSIFICATION = "classification"
+
+TASK_TYPES = [NomicAITaskType.SEARCH_QUERY, NomicAITaskType.SEARCH_DOCUMENT, NomicAITaskType.CLUSTERING, NomicAITaskType.CLASSIFICATION]
+
+
+class NomicEmbedding(BaseEmbedding):
+    """NomicEmbedding uses the Nomic API to generate embeddings."""
+
+    # Instance variables initialized via Pydantic's mechanism
+    task_type: Optional[str] = Field(description="Embedding prefix")
+    model_name: Optional[str] = Field(description="Embedding model name")
+    _model: Any = PrivateAttr()
+
+    def __init__(
+        self,
+        model_name: str = "nomic-text-embed-v1",
+        embed_batch_size: int = 32,
+        api_key: Optional[str] = None,
+        callback_manager: Optional[CallbackManager] = None,
+        task_type: Optional[str] = "search_document",
+        **kwargs: Any,
+    ) -> None:
+        if task_type not in TASK_TYPES:
+            raise ValueError(f"Invalid task type {task_type}. Must be one of {TASK_TYPES}")
+
+        try:
+            import nomic
+            from nomic import embed
+        except ImportError:
+            raise ImportError(
+                "NomicEmbedding requires the 'nomic' package to be installed.\n"
+                "Please install it with `pip install nomic`."
+            )
+
+        if api_key is not None:
+            nomic.cli.login(api_key)
+        super().__init__(
+            embed_batch_size=embed_batch_size,
+            callback_manager=callback_manager,
+            _model=embed,
+            task_type=task_type,
+            **kwargs,
+        )
+        self._model = embed
+        self.model_name = model_name
+        self.task_type = task_type
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "NomicEmbedding"
+
+    def _embed(self, texts: List[str], task_type: Optional[str] = None) -> List[List[float]]:
+        """Embed sentences using NomicAI."""
+        result = self._model.text(texts, model=self.model_name, task_type=self.task_type)
+        return result["embeddings"]
+
+    def _get_query_embedding(self, query: str) -> List[float]:
+        """Get query embedding."""
+        return self._embed([query])[0]
```

Comment:
should get_text_embedding and get_query_embedding be using the appropriate modes automatically?